### PR TITLE
new console test for ensuring that only valid addressTypes and Plans …

### PR DESCRIPTION
**Pull request by RH-sdavey**
…are able to be selected

    * doTestAddressWithValidPlanOnly
    * new methods for selecting various WebElements in the create address modal window

### CODE REVIEW

File: systemtests/src/main/java/io/enmasse/systemtest/selenium/page/ConsoleWebPage.java [hunk 1]
```diff
@@ -317,6 +313,54 @@ private WebElement getFilterRegexAlertClose() throws Exception {
         return selenium.getWebElement(() -> selenium.getDriver().findElement(By.className("pficon-close")));
     }
 
+    //
+    //Create address modal window getters
+    //
+
+    public WebElement getAddressModalNextButton() throws Exception {
```
Comment (kornys) [inline systemtests/src/main/java/io/enmasse/systemtest/selenium/page/ConsoleWebPage.java]: redundant, ConsoleWebPage already contains getNextButton

File: systemtests/src/main/java/io/enmasse/systemtest/selenium/page/ConsoleWebPage.java [hunk 2]
```diff
@@ -317,6 +313,54 @@ private WebElement getFilterRegexAlertClose() throws Exception {
...
+        return selenium.getWebElement(() -> selenium.getDriver().findElement(By.id("nextButton")));
+    }
+
+    /**
+     * get Address Modal window Back button
+     */
+    private WebElement getAddressModalBackButton() throws Exception {
```
Comment (kornys) [inline systemtests/src/main/java/io/enmasse/systemtest/selenium/page/ConsoleWebPage.java]: same as above, I implemented these methods long time ago, you don't need to create same methods with different name

File: systemtests/src/test/java/io/enmasse/systemtest/bases/web/WebConsoleTest.java [hunk 3]
```diff
@@ -806,6 +806,44 @@ protected void doTestCreateAddressWithSymbolsAt61stCharIndex(Destination... dest
...
+
+        // skip straight back to page 3 and create address
+        consoleWebPage.clickOnAddressModalPageByNumber(3);
+        consoleWebPage.clickOnAddressModalNextButton();
+
+        // assert new address is Topic
+        assertEquals(AddressType.TOPIC.toString(), consoleWebPage.getAddressItem(destTopic).getType(),
+                "Console failed, expected TOPIC type");
+
+        assertCanConnect(sharedAddressSpace, defaultCredentials, Collections.singletonList(destTopic));
```
[the remark this answers is not in the document]
Comment (RH-sdavey) [reply systemtests/src/test/java/io/enmasse/systemtest/bases/web/WebConsoleTest.java]: waitForDestinationsReady(sharedAddressSpace, destTopic);   added

File: systemtests/src/main/java/io/enmasse/systemtest/selenium/page/ConsoleWebPage.java (hunk 1 above)
Comment (RH-sdavey) [reply systemtests/src/main/java/io/enmasse/systemtest/selenium/page/ConsoleWebPage.java]: removed

File: systemtests/src/main/java/io/enmasse/systemtest/selenium/page/ConsoleWebPage.java (hunk 2 above)
Comment (RH-sdavey) [reply systemtests/src/main/java/io/enmasse/systemtest/selenium/page/ConsoleWebPage.java]: removed

File: systemtests/src/main/java/io/enmasse/systemtest/selenium/page/ConsoleWebPage.java [hunk 4]
```diff
@@ -317,6 +313,54 @@ private WebElement getFilterRegexAlertClose() throws Exception {
...
+
+    public WebElement getAddressModalPageByNumber(Integer pageNumber) throws Exception {
+        return getAddressModalPageNumbers().findElements(By.className("wizard-pf-step-number")).get(pageNumber - 1);  //zero indexed
+    }
+
+    public WebElement getCreateAddressModalWindow() throws Exception {
+        return selenium.getDriver().findElement(By.className("modal-dialog")).findElement(By.className("modal-content"));
+    }
+
+    public WebElement getAddressNameInputBox() throws Exception {
```
[the remark this answers is not in the document]
Comment (RH-sdavey) [reply systemtests/src/main/java/io/enmasse/systemtest/selenium/page/ConsoleWebPage.java]: removed

File: systemtests/src/main/java/io/enmasse/systemtest/selenium/page/ConsoleWebPage.java [hunk 5]
```diff
@@ -317,6 +313,54 @@ private WebElement getFilterRegexAlertClose() throws Exception {
...
+     */
+    private WebElement getAddressModalPageNumbers() throws Exception {
+        return selenium.getDriver().findElement(By.className("wizard-pf-steps-indicator"));
+    }
+
+    public WebElement getAddressModalPageByNumber(Integer pageNumber) throws Exception {
+        return getAddressModalPageNumbers().findElements(By.className("wizard-pf-step-number")).get(pageNumber - 1);  //zero indexed
+    }
+
+    public WebElement getCreateAddressModalWindow() throws Exception {
```
[the remark this answers is not in the document]
Comment (RH-sdavey) [reply systemtests/src/main/java/io/enmasse/systemtest/selenium/page/ConsoleWebPage.java]: removed

File: systemtests/src/main/java/io/enmasse/systemtest/selenium/page/ConsoleWebPage.java [hunk 6]
```diff
@@ -375,6 +419,22 @@ public void clickOnRemoveButton() throws Exception {
         selenium.clickOnItem(getRemoveButton());
     }
 
+    public void clickOnAddressModalNextButton() throws Exception {
```
[the remark this answers is not in the document]
Comment (RH-sdavey) [reply systemtests/src/main/java/io/enmasse/systemtest/selenium/page/ConsoleWebPage.java]: removed

File: systemtests/src/main/java/io/enmasse/systemtest/selenium/page/ConsoleWebPage.java [hunk 7]
```diff
@@ -375,6 +419,22 @@ public void clickOnRemoveButton() throws Exception {
...
+        selenium.clickOnItem(getAddressModalNextButton());
+    }
+
+    public void clickOnAddressModalBackButton() throws Exception {
```
[the remark this answers is not in the document]
Comment (RH-sdavey) [reply systemtests/src/main/java/io/enmasse/systemtest/selenium/page/ConsoleWebPage.java]: removed

**Pull request merged**